frontend/pages/events.js

```diff
@@ -0,0 +1,361 @@
+  const setTerms = (events) => {
+    const tempPastEvents = {
+      term1: events.filter((event) => event.term === 1),
+      term2: events.filter((event) => event.term === 2),
+      term3: events.filter((event) => event.term === 3),
+    };
+    setPastEvents(tempPastEvents);
+    setPastSelectedPosts(tempPastEvents);
+    setPastContent(tempPastEvents);
+    setLoadingPast(false);
+  };
+
+  // events archive message
+  useEffect(() => {
+    const isEmpty = Object.values(pastEvents).every(
+      (term) => term.length === 0
+    );
+
+    if (isEmpty && loadingPast === false) {
+      setEmptyCategory(true);
+      console.error = () => {};
+    } else {
+      setEmptyCategory(false);
+    }
+  }, [pastEvents, loadingPast]);
+
+  // filter past events of a year by selected category
+  const filterContent = (selectedCategory) => {
+    const filteredContent = {
+      term1: filterTerm("term1", selectedCategory),
+      term2: filterTerm("term2", selectedCategory),
+      term3: filterTerm("term3", selectedCategory),
+    };
+    setPastSelectedPosts(filteredContent);
+    setPastEvents(filteredContent);
+  };
+
+  // filter the past events of the given term by the selected category
+  const filterTerm = (term, selectedCategory) => {
+    const filteredTerm = pastContent[term].filter(
+      (picture) =>
+        selectedCategory === "All" ||
+        picture.category.split(",").includes(selectedCategory)
+    );
+
+    return filteredTerm;
+  };
+
+  // get upcoming events
+  // input: upcoming events data from google sheets
+  // output: array of dictionaries containing upcoming events data
+  const fetchUpcomingEvents = async () => {
+    const res = await axios.get(
+      "https://wit-database.herokuapp.com/upcoming-events"
+    );
+    const tempEvents = humps.camelizeKeys(res.data);
+    setUpcomingEvents(tempEvents);
+    setCurrentPosts(tempEvents.slice(0, postsPerPage));
+    setSelectedPosts(tempEvents);
+    setLoadingUpcoming(false);
+    setSourceLoading(false);
+  };
+
+  // load upcoming events
+  useEffect(() => {
+    setLoadingUpcoming(true);
+    fetchUpcomingEvents().catch((error) =>
+      // error handling
+      console.error(error)
+    );
+  }, []);
+
+  // get events for a specific term
+  const getTermEvents = (events) => {
+    return events.map((event, index) => {
+      let eventLabel = event.img.split(".")[0].split("-");
+      eventLabel.shift();
+      let event_id = `${year}_${event.eventNumber}`;
+      return (
+        <div className={styles.pastEvent} key={index}>
+          <Link href={`/event-recaps/${event_id}`}>
+            <div className={styles.eventImgBox}>
+              <Image
+                className={styles.eventImages}
+                src={`/event-covers/${year}/${event.img}`}
+                alt={eventLabel.join(" ")}
+                width="1200px"
+                height="628px"
+              />
+              <div className={styles.darkOverlay} />
```

Comment:
Fix the dimensions of the overlay - think it's slightly off 
![image](https://user-images.githubusercontent.com/80564580/176904212-e5376e3c-327f-4e34-b953-6f6c4aeb1a6e.png)

Reply:
done !!